File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/games/tools/ToolRouter.kt
package com.wagerproof.app.features.games.tools

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.wagerproof.app.features.analytics.MlbRegressionReportScreen
import com.wagerproof.app.features.outliers.NBAModelAccuracyView
import com.wagerproof.app.features.outliers.NCAABModelAccuracyView
import com.wagerproof.core.stores.OutliersStore

/**
 * Maps a tool's [OutliersStore.Category] to its leaf page. iOS
 * `Games/Tools/ToolRouter`. Both entry points (Games banner + Outliers hub)
 * call this so they open the identical view. `value`/`fade` aren't tools.
 */
object ToolRouter {
    @Composable
    fun LeafView(category: OutliersStore.Category, modifier: Modifier = Modifier) {
        when (category) {
            OutliersStore.Category.nbaAccuracy -> NBAModelAccuracyView(modifier)
            OutliersStore.Category.ncaabAccuracy -> NCAABModelAccuracyView(modifier)
            OutliersStore.Category.mlbRegression -> MlbRegressionReportScreen(modifier)
            OutliersStore.Category.`value`, OutliersStore.Category.fade -> Unit
        }
    }
}

File: wagerproof-ios-native/Wagerproof/Features/Games/Tools/ToolRouter.swift
import SwiftUI
import WagerproofStores

/// Single source of truth mapping a tool `OutliersStore.Category` to its leaf
/// page. Both entry points — the Games-page banner push and the Outliers hub
/// detail router — call this so they always open the identical (restyled) view.
/// `.value`/`.fade` are not tools; OutliersDetailView renders those inline.
enum ToolRouter {
    @ViewBuilder
    static func leafView(for category: OutliersStore.Category) -> some View {
        switch category {
        case .nbaAccuracy:        NBAModelAccuracyView()
        case .ncaabAccuracy:      NCAABModelAccuracyView()
        case .mlbRegression:      MlbRegressionReportView()
        case .nflHistoricalAnalysis: HistoricalAnalysisView(sport: .nfl)
        case .cfbHistoricalAnalysis: HistoricalAnalysisView(sport: .cfb)
        case .value, .fade:       EmptyView()
        }
    }
}
